lib/jiff-client.js

```diff
@@ -3032,8 +3066,12 @@
      * @member {number} Zp
      * @instance
      */
-    jiff.Zp = (options.Zp == null ? gZp : options.Zp);
-
+    if (options.safemod == true || options.safemod == null) {
+      jiff.Zp = (is_prime(options.Zp) ? options.Zp : gZp);
```

Comment:
This recovers silently and the developer never realizes something is wrong, they may go about developing as if Zp is what they think they set it to, but in reality it is gZp.

Can you modify this to make it fail explicitly: so if is_prime(options.Zp) is false, you throw an exception/error with some meaningful error message? If you have not seen throwing exceptions before, this link will be interesting to read.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw

P.S. please use throw new Error('...message..') and not the throw 'message'. When the first is logged to the screen/console, the user sees a trace of where the error happened and other useful information, while the later only displays the message without any additional information.